openedx/core/djangoapps/content/course_overviews/signals.py

```diff
@@ -11,9 +11,10 @@
 def _listen_for_course_publish(sender, course_key, **kwargs):  # pylint: disable=unused-argument
     """
     Catches the signal that a course has been published in Studio and
-    invalidates the corresponding CourseOverview cache entry if one exists.
+    updates the corresponding CourseOverview cache entry.
     """
     CourseOverview.objects.filter(id=course_key).delete()
+    CourseOverview.load_from_module_store(course_key)
```

Comment:
My main concern around this change is this re-population line. Although this work is done asynchronously, if multiple publish signals are being triggered for some reason (perhaps unnecessarily), it could add load where it didn't exist before. Those who are more familiar with the course publish signal are free to tell me that my concerns are misplaced - I'd welcome it! Not a blocker - just a concern.